core/src/ids.rs

```diff
@@ -206,3 +206,19 @@ impl ProgramId {
         hash(&argument).into()
     }
 }
+
+declare_id!(ReservationId: "Reservation identifier");
```

Comment:
As I said before, I don't mean `ReservationId` as a common and **core** part of  the code. As I don't mean `Handle` in `gcore`, for example

I'd operate with just `type Hash = [u8; 32];` in core part, defining some `struct ReservationId` only for contracts (user-space)

Reply:
Then `GasNodeId` will be 
```
enum GasNodeId {
     MessageId(MessageId),
     ReservationId(Hash),
}
```

Reply:
> As I said before, I don't mean ReservationId as a common and core part of the code. As I don't mean Handle in gcore, for example

`ReservationId` will be used in `Ext` trait methods so I don't want to operate with just `Hash`